pkg/pipeline/extract/timebased/filters.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package timebased
+
+import (
+	"container/list"
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	log "github.com/sirupsen/logrus"
+)
+
+func (fs *FilterStruct) CalculateResults(nowInSecs int64) {
+	log.Debugf("CalculateResults nowInSecs = %d", nowInSecs)
+	oldestValidTime := nowInSecs - int64(fs.Rule.TimeInterval)
+	for key, l := range fs.RecordKeyDataTable.dataTableMap {
+		var valueFloat64 = float64(0)
+		switch fs.Rule.Operation {
+		case OperationLast:
+			// handle empty list
+			if l.Len() == 0 {
+				continue
+			}
+			cEntry := l.Back().Value.(*TableEntry)
+			valueString := fmt.Sprintf("%v", cEntry.entry[fs.Rule.OperationKey])
+			valueFloat64, _ = strconv.ParseFloat(valueString, 64)
+		case OperationDiff:
+			for e := l.Front(); e != nil; e = e.Next() {
+				cEntry := e.Value.(*TableEntry)
+				if cEntry.timeStamp < oldestValidTime {
+					// entry is out of time range; ignore it
+					continue
+				}
+				valueString := fmt.Sprintf("%v", e.Value.(*TableEntry).entry[fs.Rule.OperationKey])
+				first, _ := strconv.ParseFloat(valueString, 64)
+				valueString = fmt.Sprintf("%v", l.Back().Value.(*TableEntry).entry[fs.Rule.OperationKey])
+				last, _ := strconv.ParseFloat(valueString, 64)
```

Comment:
Instead of converting values to string and then to float64, `utils.ConvertToFloat64()` can be used.
https://github.com/netobserv/flowlogs-pipeline/blob/6e05f32e1fa8bb22d8556b642cb3a5e788d366a4/pkg/pipeline/utils/convert.go#L15

Reply:
OK